### ISSUE TITLE
Invalid TCX format according to Garmin

### ISSUE THREAD

**Issue by felixschweizer** [philosowaffle/peloton-to-garmin#144]
I am running V2 using the windows exe, creating tcx files. After a few false starts, I get the exe to complete, but I get an upload error. Trying to upload the files manually to Garmin, I get "We don't support the file type you tried to import." I then tried to make fit files but it says "No files to convert in the download directory. Nothing to do" and "no upload directory found. Nothing to do." I am uploading one of the *.tcx files that does not upload. I have changed the extension to *.txt so it uploads
any suggestions?
[1fce507d71da47668aa11508a2309011_30_min_80s_Ride_with_Hannah_Corbin.txt](https://github.com/philosowaffle/peloton-to-garmin/files/6836894/1fce507d71da47668aa11508a2309011_30_min_80s_Ride_with_Hannah_Corbin.txt)

**Comment by philosowaffle** [comment]
For now:

1. Switch the file format to FIT
2. Find file name `synchistory.json` in the P2G directory and delete it (this file prevents P2G from processing the same workout multiple times, so if you want to convert a workout that has already been converted, you need to delete it)
3. Run the program again

When I have time again I'll take a look at why the TCX format is not working.

**Comment by felixschweizer** [comment]
Hi philosowaffle,
I did as suggested and the fit files import without any issues, as long as I do it manually. The auto-upload is unhappy and I attach a screen shot of the console. 
thanks for this amazing program!!
Felix
![2021-07-19_14-57-55](https://user-images.githubusercontent.com/6539603/126233050-826f750f-2f6e-4fd7-9201-5baaaee68f2d.jpg)

**Comment by philosowaffle** [comment]
That error is related to #109 .  Since the default upload strategy is not working, you can try modifying your config file to set the [Garmin.UploadStrategy](https://github.com/philosowaffle/peloton-to-garmin/wiki/Configuration#garmin-config) to `2`.

**Comment by felixschweizer** [comment]
I set the upload strategy to 2 as suggested and it works like a charm!

**Comment by philosowaffle** [comment]
Did a bit of investigation today on why the TCX format was not working.  Did not make much progress, but documenting my findings so far:

1. I went to strava and exported a Peloton ride directly from Strava (tcx file)
2. I then tried to manually upload this TCX file to Garmin and got the following error:

![chrome_ClpxXPGFtI](https://user-images.githubusercontent.com/7275041/126410127-4b839d29-d6c8-4774-aeb5-a893a05ea633.png)

So need to see if either Garmin has a bug on their end, or if maybe Peloton has something in the data that Garmin doesn't like.

From a cursory glance, the TCX file from P2G still looks valid, but its hard to manually check all of it.

**Comment by gavdraper** [comment]
I think I've managed to manually fix a TCX file exported with PelotonToGarm to the point that Garmin connect accepts it. There were two things I had to change, some XML NameSpaces and adding an attribute to the Creator element...

![image](https://user-images.githubusercontent.com/971016/134469708-bc9c45a0-5262-4981-8aaf-46e98fbe0ba5.png)

![image](https://user-images.githubusercontent.com/971016/134469752-dceeb2c4-e487-4e7b-9636-01d2dc1ce543.png)

I'm happy to take a look at putting a fix in, TBH I've not really looked at the code for this project so don't know if you're already doing any transforms or if you're proxying straight from Peloton to Garmin. Probably will get some time over the weekend.
